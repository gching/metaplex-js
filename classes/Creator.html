<!doctype html>
<html class="default no-js">
<head>
	<meta charset="utf-8">
	<meta http-equiv="X-UA-Compatible" content="IE=edge">
	<title>Creator | metaplex-js</title>
	<meta name="description" content="Documentation for metaplex-js">
	<meta name="viewport" content="width=device-width, initial-scale=1">
	<link rel="stylesheet" href="../assets/css/main.css">
	<script async src="../assets/js/search.js" id="search-script"></script>
</head>
<body>
<header>
	<div class="tsd-page-toolbar">
		<div class="container">
			<div class="table-wrap">
				<div class="table-cell" id="tsd-search" data-index="../assets/js/search.json" data-base="..">
					<div class="field">
						<label for="tsd-search-field" class="tsd-widget search no-caption">Search</label>
						<input id="tsd-search-field" type="text" />
					</div>
					<ul class="results">
						<li class="state loading">Preparing search index...</li>
						<li class="state failure">The search index is not available</li>
					</ul>
					<a href="../index.html" class="title">metaplex-js</a>
				</div>
				<div class="table-cell" id="tsd-widgets">
					<div id="tsd-filter">
						<a href="#" class="tsd-widget options no-caption" data-toggle="options">Options</a>
						<div class="tsd-filter-group">
							<div class="tsd-select" id="tsd-filter-visibility">
								<span class="tsd-select-label">All</span>
								<ul class="tsd-select-list">
									<li data-value="public">Public</li>
									<li data-value="protected">Public/Protected</li>
									<li data-value="private" class="selected">All</li>
								</ul>
							</div>
							<input type="checkbox" id="tsd-filter-inherited" checked />
							<label class="tsd-widget" for="tsd-filter-inherited">Inherited</label>
							<input type="checkbox" id="tsd-filter-externals" checked />
							<label class="tsd-widget" for="tsd-filter-externals">Externals</label>
						</div>
					</div>
					<a href="#" class="tsd-widget menu no-caption" data-toggle="menu">Menu</a>
				</div>
			</div>
		</div>
	</div>
	<div class="tsd-page-title">
		<div class="container">
			<ul class="tsd-breadcrumb">
				<li>
					<a href="../modules.html">metaplex-js</a>
				</li>
				<li>
					<a href="Creator.html">Creator</a>
				</li>
			</ul>
			<h1>Class Creator</h1>
		</div>
	</div>
</header>
<div class="container container-main">
	<div class="row">
		<div class="col-8 col-content">
			<section class="tsd-panel tsd-hierarchy">
				<h3>Hierarchy</h3>
				<ul class="tsd-hierarchy">
					<li>
						<span class="target">Creator</span>
					</li>
				</ul>
			</section>
			<section class="tsd-panel-group tsd-index-group">
				<h2>Index</h2>
				<section class="tsd-panel tsd-index-panel">
					<div class="tsd-index-content">
						<section class="tsd-index-section ">
							<h3>Constructors</h3>
							<ul class="tsd-index-list">
								<li class="tsd-kind-constructor tsd-parent-kind-class"><a href="Creator.html#constructor" class="tsd-kind-icon">constructor</a></li>
							</ul>
						</section>
						<section class="tsd-index-section ">
							<h3>Properties</h3>
							<ul class="tsd-index-list">
								<li class="tsd-kind-property tsd-parent-kind-class"><a href="Creator.html#address" class="tsd-kind-icon">address</a></li>
								<li class="tsd-kind-property tsd-parent-kind-class"><a href="Creator.html#share" class="tsd-kind-icon">share</a></li>
								<li class="tsd-kind-property tsd-parent-kind-class"><a href="Creator.html#verified" class="tsd-kind-icon">verified</a></li>
								<li class="tsd-kind-property tsd-parent-kind-class tsd-is-static"><a href="Creator.html#SCHEMA" class="tsd-kind-icon">SCHEMA</a></li>
							</ul>
						</section>
						<section class="tsd-index-section ">
							<h3>Accessors</h3>
							<ul class="tsd-index-list">
								<li class="tsd-kind-get-signature tsd-parent-kind-class"><a href="Creator.html#isVerified" class="tsd-kind-icon">is<wbr>Verified</a></li>
							</ul>
						</section>
					</div>
				</section>
			</section>
			<section class="tsd-panel-group tsd-member-group ">
				<h2>Constructors</h2>
				<section class="tsd-panel tsd-member tsd-kind-constructor tsd-parent-kind-class">
					<a name="constructor" class="tsd-anchor"></a>
					<h3>constructor</h3>
					<ul class="tsd-signatures tsd-kind-constructor tsd-parent-kind-class">
						<li class="tsd-signature tsd-kind-icon">new <wbr>Creator<span class="tsd-signature-symbol">(</span>args<span class="tsd-signature-symbol">: </span><span class="tsd-signature-symbol">{ </span>address<span class="tsd-signature-symbol">: </span><span class="tsd-signature-type">PublicKey</span><span class="tsd-signature-symbol">; </span>share<span class="tsd-signature-symbol">: </span><span class="tsd-signature-type">number</span><span class="tsd-signature-symbol">; </span>verified<span class="tsd-signature-symbol">: </span><span class="tsd-signature-type">0</span><span class="tsd-signature-symbol"> | </span><span class="tsd-signature-type">1</span><span class="tsd-signature-symbol"> }</span><span class="tsd-signature-symbol">)</span><span class="tsd-signature-symbol">: </span><a href="Creator.html" class="tsd-signature-type" data-tsd-kind="Class">Creator</a></li>
					</ul>
					<ul class="tsd-descriptions">
						<li class="tsd-description">
							<aside class="tsd-sources">
								<ul>
									<li>Defined in <a href="https://github.com/gching/metaplex-js/blob/d2c3ee6/src/models/programs/token-metadata/Creator.ts#L27">models/programs/token-metadata/Creator.ts:27</a></li>
								</ul>
							</aside>
							<h4 class="tsd-parameters-title">Parameters</h4>
							<ul class="tsd-parameters">
								<li>
									<h5>args: <span class="tsd-signature-symbol">{ </span>address<span class="tsd-signature-symbol">: </span><span class="tsd-signature-type">PublicKey</span><span class="tsd-signature-symbol">; </span>share<span class="tsd-signature-symbol">: </span><span class="tsd-signature-type">number</span><span class="tsd-signature-symbol">; </span>verified<span class="tsd-signature-symbol">: </span><span class="tsd-signature-type">0</span><span class="tsd-signature-symbol"> | </span><span class="tsd-signature-type">1</span><span class="tsd-signature-symbol"> }</span></h5>
									<ul class="tsd-parameters">
										<li class="tsd-parameter">
											<h5>address<span class="tsd-signature-symbol">: </span><span class="tsd-signature-type">PublicKey</span></h5>
										</li>
										<li class="tsd-parameter">
											<h5>share<span class="tsd-signature-symbol">: </span><span class="tsd-signature-type">number</span></h5>
										</li>
										<li class="tsd-parameter">
											<h5>verified<span class="tsd-signature-symbol">: </span><span class="tsd-signature-type">0</span><span class="tsd-signature-symbol"> | </span><span class="tsd-signature-type">1</span></h5>
										</li>
									</ul>
								</li>
							</ul>
							<h4 class="tsd-returns-title">Returns <a href="Creator.html" class="tsd-signature-type" data-tsd-kind="Class">Creator</a></h4>
						</li>
					</ul>
				</section>
			</section>
			<section class="tsd-panel-group tsd-member-group ">
				<h2>Properties</h2>
				<section class="tsd-panel tsd-member tsd-kind-property tsd-parent-kind-class">
					<a name="address" class="tsd-anchor"></a>
					<h3>address</h3>
					<div class="tsd-signature tsd-kind-icon">address<span class="tsd-signature-symbol">:</span> <span class="tsd-signature-type">PublicKey</span></div>
					<aside class="tsd-sources">
						<ul>
							<li>Defined in <a href="https://github.com/gching/metaplex-js/blob/d2c3ee6/src/models/programs/token-metadata/Creator.ts#L19">models/programs/token-metadata/Creator.ts:19</a></li>
						</ul>
					</aside>
				</section>
				<section class="tsd-panel tsd-member tsd-kind-property tsd-parent-kind-class">
					<a name="share" class="tsd-anchor"></a>
					<h3>share</h3>
					<div class="tsd-signature tsd-kind-icon">share<span class="tsd-signature-symbol">:</span> <span class="tsd-signature-type">number</span></div>
					<aside class="tsd-sources">
						<ul>
							<li>Defined in <a href="https://github.com/gching/metaplex-js/blob/d2c3ee6/src/models/programs/token-metadata/Creator.ts#L21">models/programs/token-metadata/Creator.ts:21</a></li>
						</ul>
					</aside>
				</section>
				<section class="tsd-panel tsd-member tsd-kind-property tsd-parent-kind-class">
					<a name="verified" class="tsd-anchor"></a>
					<h3>verified</h3>
					<div class="tsd-signature tsd-kind-icon">verified<span class="tsd-signature-symbol">:</span> <span class="tsd-signature-type">0</span><span class="tsd-signature-symbol"> | </span><span class="tsd-signature-type">1</span></div>
					<aside class="tsd-sources">
						<ul>
							<li>Defined in <a href="https://github.com/gching/metaplex-js/blob/d2c3ee6/src/models/programs/token-metadata/Creator.ts#L20">models/programs/token-metadata/Creator.ts:20</a></li>
						</ul>
					</aside>
				</section>
				<section class="tsd-panel tsd-member tsd-kind-property tsd-parent-kind-class tsd-is-static">
					<a name="SCHEMA" class="tsd-anchor"></a>
					<h3><span class="tsd-flag ts-flagStatic">Static</span> SCHEMA</h3>
					<div class="tsd-signature tsd-kind-icon">SCHEMA<span class="tsd-signature-symbol">:</span> <span class="tsd-signature-type">BorshSchema</span><span class="tsd-signature-symbol"> = ...</span></div>
					<aside class="tsd-sources">
						<ul>
							<li>Defined in <a href="https://github.com/gching/metaplex-js/blob/d2c3ee6/src/models/programs/token-metadata/Creator.ts#L5">models/programs/token-metadata/Creator.ts:5</a></li>
						</ul>
					</aside>
				</section>
			</section>
			<section class="tsd-panel-group tsd-member-group ">
				<h2>Accessors</h2>
				<section class="tsd-panel tsd-member tsd-kind-get-signature tsd-parent-kind-class">
					<a name="isVerified" class="tsd-anchor"></a>
					<h3>is<wbr>Verified</h3>
					<ul class="tsd-signatures tsd-kind-get-signature tsd-parent-kind-class">
						<li class="tsd-signature tsd-kind-icon"><span class="tsd-signature-symbol">get</span> isVerified<span class="tsd-signature-symbol">(</span><span class="tsd-signature-symbol">)</span><span class="tsd-signature-symbol">: </span><span class="tsd-signature-type">boolean</span></li>
					</ul>
					<ul class="tsd-descriptions">
						<li class="tsd-description">
							<aside class="tsd-sources">
								<ul>
									<li>Defined in <a href="https://github.com/gching/metaplex-js/blob/d2c3ee6/src/models/programs/token-metadata/Creator.ts#L23">models/programs/token-metadata/Creator.ts:23</a></li>
								</ul>
							</aside>
							<h4 class="tsd-returns-title">Returns <span class="tsd-signature-type">boolean</span></h4>
						</li>
					</ul>
				</section>
			</section>
		</div>
		<div class="col-4 col-menu menu-sticky-wrap menu-highlight">
			<nav class="tsd-navigation primary">
				<ul>
					<li class=" ">
						<a href="../modules.html">Exports</a>
					</li>
				</ul>
			</nav>
			<nav class="tsd-navigation secondary menu-sticky">
				<ul class="before-current">
				</ul>
				<ul class="current">
					<li class="current tsd-kind-class">
						<a href="Creator.html" class="tsd-kind-icon">Creator</a>
						<ul>
							<li class=" tsd-kind-constructor tsd-parent-kind-class">
								<a href="Creator.html#constructor" class="tsd-kind-icon">constructor</a>
							</li>
							<li class=" tsd-kind-property tsd-parent-kind-class">
								<a href="Creator.html#address" class="tsd-kind-icon">address</a>
							</li>
							<li class=" tsd-kind-property tsd-parent-kind-class">
								<a href="Creator.html#share" class="tsd-kind-icon">share</a>
							</li>
							<li class=" tsd-kind-property tsd-parent-kind-class">
								<a href="Creator.html#verified" class="tsd-kind-icon">verified</a>
							</li>
							<li class=" tsd-kind-property tsd-parent-kind-class tsd-is-static">
								<a href="Creator.html#SCHEMA" class="tsd-kind-icon">SCHEMA</a>
							</li>
							<li class=" tsd-kind-get-signature tsd-parent-kind-class">
								<a href="Creator.html#isVerified" class="tsd-kind-icon">is<wbr>Verified</a>
							</li>
						</ul>
					</li>
				</ul>
				<ul class="after-current">
				</ul>
			</nav>
		</div>
	</div>
</div>
<footer class="with-border-bottom">
	<div class="container">
		<h2>Legend</h2>
		<div class="tsd-legend-group">
			<ul class="tsd-legend">
				<li class="tsd-kind-constructor tsd-parent-kind-class"><span class="tsd-kind-icon">Constructor</span></li>
				<li class="tsd-kind-property tsd-parent-kind-class"><span class="tsd-kind-icon">Property</span></li>
				<li class="tsd-kind-method tsd-parent-kind-class"><span class="tsd-kind-icon">Method</span></li>
			</ul>
			<ul class="tsd-legend">
				<li class="tsd-kind-property tsd-parent-kind-class tsd-is-static"><span class="tsd-kind-icon">Static property</span></li>
				<li class="tsd-kind-method tsd-parent-kind-class tsd-is-static"><span class="tsd-kind-icon">Static method</span></li>
			</ul>
			<ul class="tsd-legend">
				<li class="tsd-kind-property tsd-parent-kind-interface"><span class="tsd-kind-icon">Property</span></li>
			</ul>
		</div>
	</div>
</footer>
<div class="container tsd-generator">
	<p>Generated using <a href="https://typedoc.org/" target="_blank">TypeDoc</a></p>
</div>
<div class="overlay"></div>
<script src="../assets/js/main.js"></script>
</body>
</html>