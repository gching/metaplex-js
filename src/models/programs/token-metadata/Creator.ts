import { PublicKey } from '@solana/web3.js';
import { BorshSchema } from '../../utils/types';

export class Creator {
  static SCHEMA: BorshSchema = new Map([
    [
      Creator,
      {
        kind: 'struct',
        fields: [
          ['address', 'pubkey'],
          ['verified', 'u8'],
          ['share', 'u8'],
        ],
      },
    ],
  ]);

  address: PublicKey;
  verified: 0 | 1;
  share: number;

  get isVerified(): boolean {
    return this.verified === 1;
  }

  constructor(args: { address: PublicKey; verified: 0 | 1; share: number }) {
    this.address = args.address;
    this.verified = args.verified;
    this.share = args.share;
  }
}
