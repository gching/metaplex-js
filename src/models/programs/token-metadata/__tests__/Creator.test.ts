import { Creator } from '../Creator';
import { PublicKey } from '@solana/web3.js';

describe('Creator', () => {
  it('should instantiate with the correct values', () => {
    const inst = new Creator({
      address: PublicKey.default,
      share: 50,
      verified: 1,
    });

    expect(inst).toMatchSnapshot();
  });
});
