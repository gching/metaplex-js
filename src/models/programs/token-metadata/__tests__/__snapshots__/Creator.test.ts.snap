// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Creator should instantiate with the correct values 1`] = `
Creator {
  "address": PublicKey {
    "_bn": "00",
  },
  "share": 50,
  "verified": 1,
}
`;
